File: scripts/new_map.rb
require 'json'

w = ARGV[0].to_i
h = ARGV[1].to_i

puts w
puts h

@obj = {
        title:  "",
        width:  w,
        height: h,
        tiles: [],
        data: []
      }

for i in 0..(w-1) do
  for j in 0..(h-1) do
    @obj[:tiles] << {x: i, y: j}
  end
end

f = File.open("../src/assets/maps/map.json", "w")

f.puts JSON.pretty_generate(@obj)

File: scripts/new_map.js
var fs = require('fs');

var w = process.argv[2];
var h = process.argv[3];

console.log(`Generating map ${w} x ${h}`);

var data = {
  title: "",
  width: w,
  height: h,
  tiles: []
}

for(var i=0;i<w;i++){
  for(var j=0;j<h;j++) {
    data.tiles.push({x:i,y:j});
  }
}

var json = JSON.stringify(data, null, '\t');
fs.writeFile('src/assets/maps/map.json', json, 'utf8', function(){
  console.log('done');
});
